### ISSUE TITLE
oidc_proto_authorization_request not set after mod_headers

### ISSUE THREAD

**Issue by iconoeugen** [pingidentity/mod_auth_openidc#304]
I run an apache with mod_auth_oid behind a LoadBalancer that works in pass-through mode which can not add the `X-Forwarded-` headers. As the Load-Balancer listens on port 443 and the apache is listening on port 8443, I have to add the `X-Forwarded-Port` to `443` using the `RequestHeader` directive to the apache configuration to get the correct URL redirect after authentication, see: https://github.com/pingidentity/mod_auth_openidc/blob/6581138b61147a3001f030193e528cd1a1d54930/src/util.c#L390

###### Environment

- mod_auth_openidc-2.3.2 - init - cjose 0.5.1, OpenSSL 1.0.2k-fips 26 Jan 2017, EC=yes, GCM=yes, Redis=yes, JQ=no
- mod_ssl/2.4.6 compiled against Server: Apache/2.4.6, Library: OpenSSL/1.0.2k
- CentOS Linux release 7.4.1708 (Core) docker image

###### Expected behaviour

The module can see the header `X-Forwarded-Port` as part of the request without `early` option.

###### Actual behaviour

The module can not see the header `X-Forwarded-Port` as part of the request without `early` option

###### Minimized example

Works:
```
RequestHeader set X-Forwarded-Port 443 early
```

Doesn't work:
```
RequestHeader set X-Forwarded-Port 443
```

**Comment by iconoeugen** [comment]
Recommendation from apache documentation:

> Early mode is designed as a test/debugging aid for developers. Directives defined using the early keyword are set right at the beginning of processing the request. This means they can be used to simulate different requests and set up test cases, but it also means that headers may be changed at any time by other modules before generating a Response.

**Comment by zandbelt** [comment]
The text also mentions:

> mod_headers can be applied either early or late in the request. The normal mode is late, when Request Headers are set immediately before running the content generator

But the content generator is, by nature, triggered only **after** the authentication/authorization phase so the normal header setting is too late for that. This behavior cannot be changed by mod_auth_openidc.The fact that this is used for testing/debugging usually and that other headers may change it afterwards is not important here, it is not unsafe/insecure or anything. In your case it is used as a stop-gap for something that a Load Balancer could/should have done and that is OK.

In summary: there's nothing mod_auth_openidc can do about it, it's not a problem in itself and there's no alternative for you.
